R/driver_rds.R

```diff
@@ -76,6 +76,16 @@
 ##'
 ##' @param default_namespace Default namespace (see
 ##'   \code{\link{storr}}).
+##'   
+##' @param mangle_key_encode Optional function for mangling keys.
```

Comment:
I am thinking about the interface here and I think that we should offer a single argument here; the existing `mangle_key` which can take string arguments

- `none` - equivalent to the current `FALSE`
- `base64` - equivalent to the current `TRUE`
- `xxxx` - a key for a "registered" mangler

Then we need a new function:

```
register_mangler <- function(name, encode, decode, overwrite = FALSE) {...}
```

which adds a new mangling strategy.  Then drake would call as part of initialisation (`.onLoad`), `storr::register_mangler("drake::lazy_base64", drake::lazy_base64_encode, drake::lazy_base64_decode)`

We'll save the functions into the class much as you have here.

My thinking here is that `storr` archives need to be openable by previous and future versions of storr, and that the options should be saved into the archive itself to avoid corrupting them (there's some fairly nasty logic going on already for that).

Reply:
I like that idea the best so far. Implementation and back compatibility are much easier this way. I will start a new branch and PR tomorrow with this.